.appveyor.yml

```diff
@@ -66,7 +66,7 @@ install:
   - activate test-environment
   - echo %PYTHON_VERSION% %TARGET_ARCH%
   # pytest-cov>=2.3.1 due to https://github.com/pytest-dev/pytest-cov/issues/124
-  - pip install -q "pytest!=3.3.0" "pytest-cov>=2.3.1" pytest-rerunfailures pytest-timeout pytest-xdist
```

Comment:
I am a bit worried about requiring the newest version of pytest?  Are there new features we need or just bug fixes and this was the easiest way to make sure we got them?

Reply:
I think this is what I needed to get `pytest.params` to work, which I used to mark the individual failing tests. I can do a rewrite to get around that, it just makes the tests even clunkier.

Reply:
This is where it's from: https://github.com/matplotlib/matplotlib/pull/9783/files#diff-f36a7b45d6a24734ba38d2da8f52f138R257

Reply:
Lets see if we get push back from the packagers.

Reply:
We mark parametrized tests as xfail without using `pytest.param` all over the place. For example, [`needs_usetex` used in this parameter](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_backend_svg.py#L134) is [really an `xfail`](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_backend_svg.py#L19).

So I'm not so sure you need to bump requirements here.

Reply:
Happy to change this:
```python
PLOT_BROKEN_LIST = [Axes.scatter,
              pytest.param(Axes.plot, marks=pytest.mark.xfail),
              pytest.param(Axes.bar, marks=pytest.mark.xfail)]
```
just dunno how. pytest.param is [how pytest suggests doing it](https://docs.pytest.org/en/latest/skipping.html#skip-xfail-with-parametrize)

Reply:
You have to go back to an old version of the docs, e.g. [3.0.0](https://docs.pytest.org/en/3.0.0/skipping.html#skip-xfail-with-parametrize).

Reply:
Thanks! Then will do that and undo the travis/appveyor changes.

Reply:
Ugh, doesn't work/fails spectacularly when I try. Technically params was introduced in 3.2, but I dunno how to say version >=3.2 or !=3.3

Reply:
Odd that `xfail` doesn't seem to catch did-not-raise errors, but Fedora has 3.2, so if that's all you need, we'd probably be fine with that. Debian other-than-stable is probably okay too.

Reply:
😄 I solved the how the specify problem: `pytest!=3.3.0, >=3.20` and travis didn't break this time. :crossed_fingers: for appveyor.